packages/hash/api/src/graphql/resolvers/pages/setParentPage.ts

```diff
@@ -0,0 +1,75 @@
+import { ApolloError } from "apollo-server-express";
+import { MutationSetParentPageArgs, Resolver } from "../../apiTypes.gen";
+import { LoggedInGraphQLContext } from "../../context";
+import { Entity, Page, UnresolvedGQLEntity } from "../../../model";
+import { topologicalSort } from "../../../util";
+
+export const setParentPage: Resolver<
+  Promise<UnresolvedGQLEntity>,
+  {},
+  LoggedInGraphQLContext,
+  MutationSetParentPageArgs
+> = async (_, { accountId, pageId, parentPageId }, { dataSources, user }) => {
+  return await dataSources.db.transaction(async (client) => {
+    const pageEntity = await Page.getAccountPageWithParents(client, {
+      accountId,
+      entityId: pageId,
+    });
+
+    const parentPageEntity = await Entity.getEntityLatestVersion(client, {
+      accountId,
+      entityId: parentPageId,
+    });
+
+    if (!pageEntity || !parentPageEntity) {
+      const notFoundId = pageEntity?.entityId
+        ? `'${parentPageId}'`
+        : `'${pageId}' ${
+            parentPageEntity?.entityId
+              ? ""
+              : `nor parent page with entityId = '${parentPageId}`
+          }`;
+      throw new ApolloError(
+        `Could not find entity with entityId = ${notFoundId}.`,
+        "NOT_FOUND",
+      );
+    }
+
+    if (pageEntity.parentEntityId) {
+      throw new Error(
+        `Page with entityId = '${pageEntity.entityId}' already has parent page entityId = '${pageEntity.parentEntityId}'`,
+      );
+    }
```

Comment:
We should permit changing the parent's page (to support moving pages around)

Reply:
Missed that functionality when implementing - my bad. Will look into that :)